### PR TITLE
FixedHash::operator++() introduced

### PR DESCRIPTION
**Pull request by gluk256**

### CODE REVIEW

File: libdevcore/FixedHash.h [hunk 1]
```diff
@@ -105,6 +105,7 @@ class FixedHash
 	FixedHash& operator&=(FixedHash const& _c) { for (unsigned i = 0; i < N; ++i) m_data[i] &= _c.m_data[i]; return *this; }
 	FixedHash operator&(FixedHash const& _c) const { return FixedHash(*this) &= _c; }
 	FixedHash operator~() const { FixedHash ret; for (unsigned i = 0; i < N; ++i) ret[i] = ~m_data[i]; return ret; }
+	FixedHash& operator++() { for (unsigned i = size; i > 0 && !++m_data[--i]; ) {}; return *this; }
```
Comment (gavofyork) [inline libdevcore/FixedHash.h]: unneeded `;` following `{}`

**Comment by gluk256** [comment]
fixed

File: test/libdevcore/FixedHash.cpp [hunk 2]
```diff
@@ -100,6 +100,42 @@ BOOST_AUTO_TEST_CASE(FixedHashContains)
...
+
+BOOST_AUTO_TEST_CASE(FixedHashIncrement)
+{
+	incrementSingleIteration(0);
+	incrementSingleIteration(1);
+	incrementSingleIteration(0xBAD);
+	incrementSingleIteration(0xBEEF);
+	incrementSingleIteration(0xFFFF);
+	incrementSingleIteration(0xFEDCBA);
+	incrementSingleIteration(0x7FFFFFFF);
```
Comment (chriseth) [inline test/libdevcore/FixedHash.cpp]: Can you add a test for `0xffffffff'?

**Comment by gluk256** [comment]
test for 0xFFFFFFFF added

**Pull request merged**